### PR TITLE
Feature/tao 7389/extend monitoring data with execution context

### PR DESCRIPTION
**Pull request by SashaTAO**
Store delivery execution context data in delivery monitoring data.

Requires: 
https://github.com/oat-sa/extension-tao-delivery/pull/385
https://github.com/oat-sa/extension-tao-lti/pull/174

### CODE REVIEW

File: model/monitorCache/implementation/DeliveryMonitoringData.php [hunk 1]
```diff
@@ -119,6 +120,34 @@ public function setDeliveryExecution(DeliveryExecution $deliveryExecution)
         $this->deliveryExecution = $deliveryExecution;
     }
 
+    /**
+     * @return DeliveryExecutionInterface
```
Comment (peetya) [inline model/monitorCache/implementation/DeliveryMonitoringData.php]: The PHPDoc of the object variable should be also updated to `DeliveryExecutionInterface`. What do you think?

File: model/monitorCache/implementation/DeliveryMonitoringData.php [hunk 2]
```diff
@@ -119,6 +120,34 @@ public function setDeliveryExecution(DeliveryExecution $deliveryExecution)
...
+     */
+    public function getDeliveryExecution()
+    {
+        return $this->deliveryExecution;
+    }
+
+    /**
+     * {@inheritdoc
```
Comment (peetya) [inline model/monitorCache/implementation/DeliveryMonitoringData.php]: Missing `}` :)

File: model/monitorCache/implementation/DeliveryMonitoringData.php [hunk 3]
```diff
@@ -119,6 +120,34 @@ public function setDeliveryExecution(DeliveryExecution $deliveryExecution)
...
+    /**
+     * {@inheritdoc}
+     */
+    public function getDeliveryExecutionContext()
+    {
+        if (isset($this->data[self::PARAM_EXECUTION_CONTEXT])) {
+            return DeliveryExecutionContext::createFromArray(json_decode($this->data[self::PARAM_EXECUTION_CONTEXT], true));
+        }
+
+        return false;
```
Comment (peetya) [inline model/monitorCache/implementation/DeliveryMonitoringData.php]: What do you think about throwing an exception?

File: test/integration/monitorCache/mock/DeliveryMonitoringData.php [hunk 4]
```diff
@@ -117,6 +120,34 @@ public function setDeliveryExecution(DeliveryExecution $deliveryExecution)
...
+    /**
+     * {@inheritdoc}
+     */
+    public function getDeliveryExecutionContext()
+    {
+        if (isset($this->data[self::PARAM_EXECUTION_CONTEXT])) {
+            return DeliveryExecutionContext::createFromArray($this->data[self::PARAM_EXECUTION_CONTEXT]);
+        }
+
+        return false;
```
Comment (peetya) [inline test/integration/monitorCache/mock/DeliveryMonitoringData.php]: The same could be applied here.

Comment (SashaTAO) [reply test/integration/monitorCache/mock/DeliveryMonitoringData.php]: It's just a fake object for test.

File: model/monitorCache/implementation/DeliveryMonitoringData.php (hunk 1 above)
Comment (SashaTAO) [reply model/monitorCache/implementation/DeliveryMonitoringData.php]: Fixed

File: model/monitorCache/implementation/DeliveryMonitoringData.php (hunk 3 above)
Comment (SashaTAO) [reply model/monitorCache/implementation/DeliveryMonitoringData.php]: As we discussed it's not a required parameter and it will be better to return `null`

**Pull request merged**